src/Microsoft.Extensions.Configuration.AzureAppConfiguration/FeatureManagement/FeatureManagementKeyValueAdapter.cs

```diff
@@ -96,5 +86,236 @@ public bool NeedsRefresh()
+                                                                else if (reader.TokenType != JsonTokenType.Null)
+                                                                {
+                                                                    throw CreateFeatureFlagFormatException(
+                                                                        FeatureManagementConstants.ParametersJsonPropertyName,
+                                                                        settingKey,
+                                                                        reader.TokenType.ToString(),
+                                                                        JsonTokenType.StartObject.ToString());
+                                                                }
+
+                                                                break;
+                                                            }
+
+                                                        default:
+                                                            break;
+                                                    }
+                                                }
+                                            }
+
+                                            featureConditions.ClientFilters.Add(clientFilter);
+                                        }
+                                    }
+                                }
+                                else
+                                {
+                                    throw CreateFeatureFlagFormatException(
+                                        FeatureManagementConstants.ClientFiltersJsonPropertyName,
+                                        settingKey,
+                                        reader.TokenType.ToString(),
+                                        JsonTokenType.StartArray.ToString());
+                                }
+
+                                break;
+                            }
+
+                        case FeatureManagementConstants.RequirementTypeJsonPropertyName:
+                            {
+                                if (reader.Read() && reader.TokenType == JsonTokenType.String)
+                                {
+                                    featureConditions.RequirementType = reader.GetString();
+                                }
+                                else
+                                {
+                                    throw CreateFeatureFlagFormatException(
+                                        FeatureManagementConstants.RequirementTypeJsonPropertyName,
+                                        settingKey,
+                                        reader.TokenType.ToString(),
+                                        JsonTokenType.String.ToString());
+                                }
+
+                                break;
+                            }
+
+                        default:
```

Comment:
What happens if there's unknown properties? I'm expecting a `Skip` of some kind is needed if an unknown property is hit.